tests/h/auth/util_test.py

```diff
@@ -200,6 +202,114 @@ def test_does_not_raise_when_authority_matches(self, pyramid_request, auth_clien
         util.validate_auth_client_authority(auth_client, authority)
 
 
+class TestPrincipalsForAuthClient(object):
+
+    def test_it_sets_auth_client_principal(self, auth_client):
+        principals = util.principals_for_auth_client(auth_client)
+
+        assert "auth_client:{authority}".format(authority=auth_client.authority) in principals
+
+    def test_it_sets_authority_principal(self, auth_client):
+        principals = util.principals_for_auth_client(auth_client)
+
+        assert "authority:{authority}".format(authority=auth_client.authority) in principals
+
+    def test_it_returns_principals_as_list(self, auth_client):
+        principals = util.principals_for_auth_client(auth_client)
+
+        assert isinstance(principals, list)
+
+
+class TestCheckAuthClient(object):
+    def test_it_queries_for_auth_client_in_db(self, pyramid_request):
+        pyramid_request.db.query.return_value.get.return_value = None
+        util.check_auth_client(username='whatever', password='random', request=pyramid_request)
+
+        pyramid_request.db.query.assert_called_once_with(AuthClient)
+        pyramid_request.db.query.return_value.get.assert_called_once_with('whatever')
+
+    def test_it_handles_sa_statement_exception_if_client_id_malformed(self, pyramid_request):
+        pyramid_request.db.query.return_value.get.side_effect = sa.exc.StatementError(
```

Comment:
This may be all kinds of ways of wrong for testing for this exception?

Reply:
I think the fragile-looking deep mocking (`pyramid_request.foo.bar.return_value.gar.side_effect = ...`) is inevitable if mocking sqlalchemy's complex API. Other than that I think this (setting the `side_effect` of a mock to an exception object) is the right way to test how something handles an exception being raised.